File: typescript/src/patterns/structural/proxy/RemotePaymentsService.ts
import { JulLogger } from '../../JulLogger.js';
import type { PaymentsService } from './PaymentsService.js';

export class RemotePaymentsService implements PaymentsService {
  private static readonly log = JulLogger.getLogger('pl.training.patterns.structural.proxy.RemotePaymentsService');

  pay(_properties: ReadonlyMap<string, string>): void {
    RemotePaymentsService.log.info('Payment started', 'pay');
  }
}

File: csharp/src/Training.Patterns/Structural/Proxy/RemotePaymentsService.cs
namespace Training.Patterns.Structural.Proxy;

public class RemotePaymentsService : IPaymentsService
{
    private static readonly JulLogger log = JulLogger.GetLogger(typeof(RemotePaymentsService));

    public void Pay(IReadOnlyDictionary<string, string> properties)
    {
        log.Info("Payment started");
    }
}
